static/js/containers/SettingsPage.js

```diff
@@ -0,0 +1,81 @@
+/* global SETTINGS: false */
+import React from 'react';
+import { connect } from 'react-redux';
+import Loader from 'react-loader';
+
+import Jumbotron from '../components/Jumbotron';
+import {
+  startProfileEdit,
+  FETCH_PROCESSING,
+  clearProfile,
+} from '../actions/index';
+import ProfileFormContainer from './ProfileFormContainer';
+
+class SettingsPage extends ProfileFormContainer {
+
+  componentDidMount() {
+    this.fetchProfile();
+  }
+
+  componentDidUpdate() {
+    this.fetchProfile();
+  }
+
+  componentWillUnmount() {
+    const { dispatch, params: { username } } = this.props;
```

Comment:
Does that make sense? We need to worry about the `username` url parameter (and check that it is or isn't the same as `SETTINGS.username`) when we're on `/users` because you can visit another user's profile by visiting `/users/some_other_users_username`.

On the settings page we're only concerned with editing the current user's profile, so all of this is ancillary. Compare how we're injecting the profile into the profile form steps (`/profile/personal`, etc) to what we're doing on `/users`.